scout/server/blueprints/panels/controllers.py

```diff
@@ -312,6 +313,46 @@ def panel_export(store, panel_obj):
     return dict(panel=panel_obj)
 
 
+def panel_export_case_hits(panel_id, institute_obj, case_obj):
+    """Fetch information required to populate the PDF report containing
+    info on all variants for a case that hit the genes of a panel
+
+    Args:
+        panel_id(str): _id of a gene panel
+        institute_obj(dict): scout.models.Institute
+        case_obj(dict): scout.models.Case
+
+    Returns:
+        data(dict): dictionary containing data to be displayed on PDF report
+    """
+    panel_obj = store.panel(panel_id)
+    panel_obj["name_and_version"] = "{}({})".format(panel_obj["display_name"], panel_obj["version"])
+    data = {"institute": institute_obj, "case": case_obj, "panel": panel_obj}
+    variant_categories = {"snv": set(), "sv": set(), "str": set(), "smn": None}
+    variants_query = {
+        "case_id": case_obj["_id"],
+        "category": None,
+        "hgnc_ids": None,
+    }
+
+    # Add variants found on panel genes info
+    for gene in panel_obj.get("genes", []):
+        variants_query["hgnc_ids"] = gene["hgnc_id"]
+        for cat, _ in variant_categories.items():
+            variants_query["category"] = cat
+            res = store.variant_collection.find_one(variants_query)
+            if res:
+                variant_categories[cat].add(gene["symbol"])
```

Comment:
I would add all genes on the panel, but do the result check only for STR?

Reply:
Agreed!